### DISCUSSION TITLE
Issues with ProScan

### DISCUSSION THREAD

**Discussion by cpg178** [chuot/rdio-scanner#239] [General]
I have noticied that ProScan creates the file upon the start of a recording and appends to it until the recording stops. The second rdio-scanner sees a new file it grabs it, but it cuts off anything after due to duplicate call detection. Increasing the wait time fixes the issue with short transmissions, but longer ones still get cut off. Disabling duplicate call detection also fixes the issue but then there are 2 of everything (1 thats cut off, and 1 full length). Am I doing something wrong or is there anyway to fix this? I guess I could crank the dwell time before ingestion up to like 30 seconds but then that means short little 2 second calls are stuck in the pipeline waiting.

This doesn't happen with my SDRTrunk DirWatch, I assume it doesn't write the audio file until it finishes processing it.

**Comment by chuot** [comment]
dirwatch will trigger as soon as a new file is created or being written to. dirwatch will then wait for 2 seconds (2000 ms) and if there's been no more writes to the file, it will then process the audio file. Did you try to boot the dirwatch delay from 2000 ms to 4000 ms to see if it solves your issue?

**Reply by cpg178** [reply]
I temporarily fixed it by bumping up the delay to 60000ms, as I had it at 10 and 20 but calls were still getting cut off. I am not sure how the ProScan program writes the files but I am not sure if its constant as rdio-scanner latches onto them as soon as the delay wears off even if the recording is still in progress.  I just tried setting it down to 4000ms but all it does is cut the audio off after 4 seconds, even if the recording is still ocurring, then when it tries to delete the file it errors saying the file is in use. 
![image](https://user-images.githubusercontent.com/4503742/179569426-5f57bbe9-a683-4151-9650-220b3af97d85.png)

**Reply by chuot** [reply]
Is everything unchecked under ProScan / Options / Recorder as per default options? Maybe some options are interfering with the way files are saved. A dirwatch.delay of 60000ms is way too long and shouldn't be that high for normal operations.

Unfortunately I don't use ProScan in my setup, but hopefully someone else using this software can shed some light on this particular issue.

**Reply by cpg178** [reply]
Yes all options are default, however I did find a work around to this. ProScan has an option to append the file name at the completion of recording with "TO (Date/Time)" so I wrote a simple script to look for files with "TO" in the name and move them to another folder where I have the dirwatch set to. Works fine with 2000ms now.

**Reply by chuot** [reply]
But still, I'm sure it is working for some peoples and I would like to understand why it isn't working for you. Maybe something changed with ProScan at some point?

I've also kindly asked the ProScan developper if it could consider using the Rdio Scanner's API documented in `docs/api.md`, but he declined unfortunately.

**Comment by cpg178** [comment]
Yes, I had inquired if there was any setting to change the MP3 recording to
make it similar to SDRTrunk and he told me there is not and it will not be
immediately implemented. I’d be interested to find out the cause as-well,
but in the meantime my fix is working.

On Mon, Aug 8, 2022 at 10:15 AM Chrystian Huot ***@***.***>
wrote:

> But still, I'm sure it is working for some peoples and I would like to
> understand why it isn't working for you. Maybe something changed with
> ProScan at some point?
>
> I've also kindly asked the ProScan developper if it could consider using
> the Rdio Scanner's API documented in docs/api.md, but he declined
> unfortunately.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/chuot/rdio-scanner/discussions/239#discussioncomment-3349942>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABCLRPXBEWNRJIOOHFYUKTLVYEJBNANCNFSM53WCZC4A>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
Sent from Gmail for iPhone
